### PR TITLE
Refactor build process, fix helm chart, use mongo-tools, cleanup on fail

### LINKED ISSUE #123: mgob is not working with mongo 4.4
Hello, 
mgob was working fine with the mongo 4.2, but i updated for 4.4 and it stopped to work.
I also update the mgob to 1.2. 

The error
/root $ curl -X POST http://localhost:8090/backup/database
{"error":"mongodump log 2020-10-30T16:34:31.137+0000\tFailed: error creating intents to dump: error creating intents for database config: error counting config.system.indexBuilds: (Unauthorized) not authorized on config to execute command { count: \"system.indexBuilds\", lsid: { id: UUID(\"74bb8d81-3f5c-4071-8a47-9239dd226336\") }, $clusterTime: { clusterTime: Timestamp(1604075661, 1), signature: { hash: BinData(0, 9DE2F8D0093F613FE3BAEEF49BCDE778DA43C31E), keyId: 6846331182976598018 } }, $db: \"config\", $readPreference: { mode: \"primary\" } } : exit status 1"}

The user has root and backup role.
Do you know what is the issue?

### PR DESCRIPTION
**Pull request by endrec**
This PR 
- adds a simple feature to clean up partial dumps from the temp folder to save some space
- updates base image tools:
  - alpine 3.14
  - mongodb tools 100.5.1 (compatible with a great range of mongo versions, from 4.0 to 5.0)
  - gnupg 2.2.31-r0
  - google cloud sdk 370.0.0
  - azure cli 2.32.0
  - aws-cli 1.22.46
- options to enable or disable different cloud storage engines
- turn the chart back to alpine, to keep the image smaller 
- completely refactored docker image build to enable to build cloud specific versions (images: https://hub.docker.com/r/endrec/mgob/tags)

This PR incorporates #148, which in turn includes #128 and #139, which fixes a bug introduced in #135 (h flag is reserved).

Fixes #123

### CODE REVIEW

**Comment by endrec** [comment]
@stefanprodan I know you are not working with mongodb any more, so I'm not sure it even worth sending PR, but as this used to be the home of mgob for years, I believe this is still the best place to add these changes.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+ARG EN_AZURE=true
+ARG AZURE_CLI_VERSION=2.32.0
+ARG EN_GCLOUD=true
+ARG GOOGLE_CLOUD_SDK_VERSION=370.0.0
+ARG EN_GPG=true
+ARG GNUPG_VERSION="2.2.31-r0"
+ARG EN_MINIO=true
+ARG EN_RCLONE=true
+
+FROM golang:1.15 as mgob-builder
```
Comment (stefanprodan) [inline Dockerfile:13]: Go 1.15 is deprecated it should work with Go 1.17

File: Dockerfile [hunk 2]
```diff
@@ -7,24 +19,59 @@ COPY . /go/src/github.com/stefanprodan/mgob
...
+
+ARG MONGODB_TOOLS_VERSION
+
+RUN apk add --no-cache git build-base krb5-dev \
+    && git clone https://github.com/mongodb/mongo-tools.git --depth 1 --branch $MONGODB_TOOLS_VERSION
+
+WORKDIR mongo-tools
+RUN ./make build
+
+FROM alpine:3.14
```
Comment (stefanprodan) [inline Dockerfile:37]: Latest Alpine is 3.15, please use that, less CVEs.

**Review by stefanprodan — COMMENTED** [review]
We switched to Debian recently because Alpine was missing some Mongo packages. I can't tell if the new image works, maybe you can confirm that this was tested?

File: Dockerfile (hunk 2 above)
Comment (endrec) [reply Dockerfile:37]: 👍🏼

File: Dockerfile (hunk 1 above)
Comment (endrec) [reply Dockerfile:13]: 👍🏼

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @endrec 🥇

**Comment by endrec** [comment]
Thanks for the super quick merge, @stefanprodan !

**Pull request merged**